plugins/uast-kotlin-fir/src/org/jetbrains/uast/kotlin/declarations/FirKotlinUFile.kt

```diff
@@ -34,12 +36,12 @@ class FirKotlinUFile(
         sourcePsi.importDirectives.map { FirKotlinUImportStatement(it, this) }
     }
 
-    override val classes: List<UClass>
-        get() {
-            // TODO: Script
-            // TODO: Facade: getOrCreateFirLightFacade()
-            return sourcePsi.declarations.mapNotNull { (it as? KtClassOrObject)?.toUClass() }
-        }
+    override val classes: List<UClass> by lz {
+        val facadeOrScriptClass = if (sourcePsi.isScript()) sourcePsi.script?.toLightClass() else sourcePsi.findFacadeClass()
+        val facadeOrScriptUClass = facadeOrScriptClass?.toUClass()?.let { listOf(it) } ?: emptyList()
+        val classes = sourcePsi.declarations.mapNotNull { (it as? KtClassOrObject)?.toUClass() }
+        facadeOrScriptUClass + classes
+    }
```

Comment:
Now `UFile` is also in its final shape I guess. My next step would be to think of ways to simplify/unify `UFile` and `UImportStatement` at each plugin.